smithy-diff/src/main/java/software/amazon/smithy/diff/ModelDiff.java

```diff
@@ -280,8 +282,16 @@ public Result compare() {
             List<DiffEvaluator> evaluators = new ArrayList<>();
             ServiceLoader.load(DiffEvaluator.class, classLoader).forEach(evaluators::add);
             Differences differences = Differences.detect(oldModel, newModel);
+
+            // Applies suppressions and elevates event severities.
+            ValidationEventDecorator decoratorResult = new ModelBasedEventDecorator()
+                    .createDecorator(newModel)
+                    .getResult()
+                    .orElse(ValidationEventDecorator.IDENTITY);
+
             List<ValidationEvent> diffEvents = evaluators.parallelStream()
                     .flatMap(evaluator -> evaluator.evaluate(differences).stream())
+                    .map(decoratorResult::decorate)
```

Comment:
Not calling here `canDecorate()` has an implicit assumption that the resulting decorator will return always `true`, I'd think that just calling the method as done somewhere else or adding a comment to make it explicit will avoid confusing readers.

Reply:
Sure can add a comment. It doesn't matter here since the canDecorate method for IDENTITY and for ModelBasedEventDecorator always returns true.